### PR TITLE
redis: add PING support

### PR DESCRIPTION
**Pull request by danielhochman**
Signed-off-by: Daniel Hochman <danielhochman@users.noreply.github.com>

**Description**
Some libraries `PING` Redis when instatiating a client to check that the connection is live. This is feature improves compatibility with those libraries.

**Risk Level**
Low

**Testing**
Manual, Unit

**Docs Changes**
https://github.com/envoyproxy/data-plane-api/pull/546

**Release Notes**
Added.

### CODE REVIEW

**Comment by keedwards** [comment]
👍 from my side (but I'm not particularly with the code base). Thank you for adding this so quickly!

File: source/common/redis/command_splitter_impl.cc [hunk 1]
```diff
@@ -325,7 +325,25 @@ InstanceImpl::InstanceImpl(ConnPool::InstancePtr&& conn_pool, Stats::Scope& scop
 }
 
 SplitRequestPtr InstanceImpl::makeRequest(const RespValue& request, SplitCallbacks& callbacks) {
-  if (request.type() != RespType::Array || request.asArray().size() < 2) {
+  if (request.type() != RespType::Array) {
```
Comment (mattklein123) [inline source/common/redis/command_splitter_impl.cc]: Would it make more sense to do this at the proxy filter layer vs. the upstream command splitter? I could go either way but just throwing it out there.

Comment (danielhochman) [reply source/common/redis/command_splitter_impl.cc]: i don't see a compelling reason to start pushing what is effectively validation logic up to the proxy_filter. right now the proxy_filter calls decode() and throws it over the wall.

this code does need a cleanup/refactor at some point but i don't think it's worth starting in this PR. probably when we start doing replication work. happy to make changes if you end up wanting them though.

Comment (mattklein123) [reply source/common/redis/command_splitter_impl.cc]: OK fair enough. I will review tomorrow.

**Pull request merged**